### PR TITLE
feat: Add charge price settlement report prepare_for_csv function

### PR DESCRIPTION
**Pull request by johevemi**
# Description

## Pull-request quality

- [ ] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/settlement_report/settlement_report_job/domain/charge_prices/prepare_for_csv.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    csv_df = pivoted_df.select(
+        map_from_dict(CHARGE_TYPES)[F.col(DataProductColumnNames.charge_type)].alias(
+            CsvColumnNames.charge_type
+        ),
+        F.col(DataProductColumnNames.charge_owner_id).alias(
+            CsvColumnNames.charge_owner_id
+        ),
+        F.col(DataProductColumnNames.charge_code).alias(CsvColumnNames.charge_code),
+        F.col(DataProductColumnNames.resolution).alias(CsvColumnNames.resolution),
+        F.col(DataProductColumnNames.is_tax).alias(CsvColumnNames.is_tax),
```
Comment (JesperGomes) [inline source/databricks/settlement_report/settlement_report_job/domain/charge_prices/prepare_for_csv.py:82]: shouldn't tax be mapped from True/False to 1/0?

File: source/databricks/settlement_report/tests/domain/charge_prices/test_charge_prices_prepare_for_csv.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+DATAHUB_ADMINISTRATOR_ID = "1234567890123"
+SYSTEM_OPERATOR_ID = "3333333333333"
+GRID_ACCESS_PROVIDER_ID = "4444444444444"
+OTHER_ID = "9999999999999"
+DEFAULT_CALCULATION_ID_BY_GRID_AREA = {
+    default_data.DEFAULT_GRID_AREA_CODE: uuid.UUID(default_data.DEFAULT_CALCULATION_ID)
+}
+DEFAULT_TIME_ZONE = "Europe/Copenhagen"
+
+JAN_1ST = datetime(2023, 12, 31, 23)
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/domain/charge_prices/test_charge_prices_prepare_for_csv.py:32]: Consider using 'Dates' from utils.py

File: source/databricks/settlement_report/tests/domain/charge_prices/test_charge_prices_prepare_for_csv.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    result_df.show()
+
+    # Assert
+    assert result_df.count() == 1
+    result = result_df.collect()[0]
+    assert result["ENERGYPRICE1"] == default_data.DEFAULT_CHARGE_PRICE
+    for i in range(2, 26):
+        assert result[f"ENERGYPRICE{i}"] is None
+
+
```
Comment (JesperGomes) [inline source/databricks/settlement_report/tests/domain/charge_prices/test_charge_prices_prepare_for_csv.py:102]: Should we have  test with daylight saving time?

File: source/databricks/settlement_report/settlement_report_job/domain/charge_prices/prepare_for_csv.py (hunk 1 above)
Comment (johevemi) [reply source/databricks/settlement_report/settlement_report_job/domain/charge_prices/prepare_for_csv.py:82]: yes, I forgot about that

File: source/databricks/settlement_report/tests/domain/charge_prices/test_charge_prices_prepare_for_csv.py (hunk 2 above)
Comment (johevemi) [reply source/databricks/settlement_report/tests/domain/charge_prices/test_charge_prices_prepare_for_csv.py:32]: I will look into it

File: source/databricks/settlement_report/tests/domain/charge_prices/test_charge_prices_prepare_for_csv.py (hunk 3 above)
Comment (johevemi) [reply source/databricks/settlement_report/tests/domain/charge_prices/test_charge_prices_prepare_for_csv.py:102]: sure we can do that

File: source/databricks/settlement_report/tests/domain/charge_prices/test_charge_prices_prepare_for_csv.py (hunk 2 above)
Comment (johevemi) [reply source/databricks/settlement_report/tests/domain/charge_prices/test_charge_prices_prepare_for_csv.py:33]: using Dates

**Pull request merged**